lib/simplifyExpression/basicsSearch/basicRules.js

```diff
@@ -0,0 +1,224 @@
+  const newNode = rewriteNode(REDUCE_ZERO_NUMERATOR, node);
+  if (newNode) {
+    return Node.Status.nodeChanged(
+      ChangeTypes.REDUCE_ZERO_NUMERATOR, node, newNode);
+  }
+  else {
+    return Node.Status.noChange(node);
+  }
+}
+
+// e.g. 2 + 0 -> 2
+const REMOVE_ADDING_ZERO = defineRuleString('#a + 0', '#a');
+
+// e.g. 0 + 2 -> 2
+const REMOVE_ADDING_ZERO_REVERSE = defineRuleString('0 + #a', '#a');
+
+// If `node` is an addition node with 0 as one of its operands,
+// remove 0 from the operands list. Returns a Node.Status object.
+function removeAdditionOfZero(node) {
+  let newNode = null;
+  newNode = rewriteNode(REMOVE_ADDING_ZERO, node);
+  if (!newNode) {
+    newNode = rewriteNode(REMOVE_ADDING_ZERO_REVERSE, node);
+  }
+  if (newNode) {
+    return Node.Status.nodeChanged(
+      ChangeTypes.REMOVE_ADDING_ZERO, node, newNode);
+  }
+  else {
+    return Node.Status.noChange(node);
+  }
+}
+
+// e.g. x ^ 1 -> x
+const REMOVE_EXPONENT_BY_ONE = defineRuleString('#a ^ 1', '#a');
+
+// If `node` is of the form x^1, reduces it to a node of the form x.
+// Returns a Node.Status object.
+function removeExponentByOne(node) {
+  const newNode = rewriteNode(REMOVE_EXPONENT_BY_ONE, node);
+  if (newNode) {
+    return Node.Status.nodeChanged(
+      ChangeTypes.REMOVE_EXPONENT_BY_ONE, node, newNode);
+  }
+  else {
+    return Node.Status.noChange(node);
+  }
+}
+
+// e.g. 1 ^ x -> 1
+const REMOVE_EXPONENT_BASE_ONE = defineRuleString('1 ^ #a', '1');
+
+// If `node` is of the form 1^x, reduces it to a node of the form 1.
+// Returns a Node.Status object.
+function removeExponentBaseOne(node) {
+  const newNode = rewriteNode(REMOVE_EXPONENT_BASE_ONE, node);
+  if (newNode && checks.resolvesToConstant(node.args[1])) {
+    return Node.Status.nodeChanged(
+      ChangeTypes.REMOVE_EXPONENT_BASE_ONE, node, newNode);
+  }
+  else {
+    return Node.Status.noChange(node);
+  }
+}
+
+// e.g. x * -1 -> -x
+const REMOVE_MULTIPLYING_BY_NEGATIVE_ONE = defineRuleString('#a * -1', '-#a');
+
+// e.g. -1 * x -> -x
+const REMOVE_MULTIPLYING_BY_NEGATIVE_ONE_REVERSE = defineRuleString('-1 * #a', '-#a');
+
+// If `node` is a multiplication node with -1 as one of its operands,
+// and a non constant as the next operand, remove -1 from the operands
+// list and make the next term have a unary minus.
+// Returns a Node.Status object.
+function removeMultiplicationByNegativeOne(node) {
+  let newNode = null;
+  newNode = rewriteNode(REMOVE_MULTIPLYING_BY_NEGATIVE_ONE, node);
+  if (!newNode) {
+    newNode = rewriteNode(REMOVE_MULTIPLYING_BY_NEGATIVE_ONE_REVERSE, node);
+  }
+  if (newNode) {
+    return Node.Status.nodeChanged(
+      ChangeTypes.REMOVE_MULTIPLYING_BY_NEGATIVE_ONE, node, newNode);
+  }
+  else {
+    return Node.Status.noChange(node);
+  }
+}
+
+// e.g. x * 1 -> x
+const REMOVE_MULTIPLYING_BY_ONE = defineRuleString('#a * 1', '#a');
+
+// e.g. 1 * x -> x
+const REMOVE_MULTIPLYING_BY_ONE_REVERSE = defineRuleString('1 * #a', '#a');
```

Comment:
do you think math-rules could implement something that did this reverse thing? 

e.g. defineRuleStringWithReverse('#a * 1', '1 * #a', '#a')

I think that'd be a bit cleaner :) looks like it might come up a lot